neurons/validators/penalty/streaming_penalty.py

```diff
@@ -29,23 +46,41 @@ def calculate_penalties(
                 streamed_text_chunks.extend(chunks)
 
             if not streamed_text_chunks:
-                accumulated_penalties[index] = 1
+                # If no chunks received, apply maximum penalty
+                accumulated_penalties[index] = 1.0
                 continue
 
+            # Token-Based Penalty Calculation
+            token_penalty = 0.0
             token_counts = [
                 (len(encoding.encode(chunk)) if chunk is not None else 0)
                 for chunk in streamed_text_chunks
             ]
 
-            # Apply penalty for exceeding max tokens per chunk
             for token_count in token_counts:
                 if token_count > MAX_TOKENS_PER_CHUNK:
-                    penalty = (
-                        token_count - MAX_TOKENS_PER_CHUNK
-                    ) * PENALTY_PER_EXCEEDING_TOKEN
+                    penalty = (token_count - MAX_TOKENS_PER_CHUNK) * PENALTY_PER_EXCEEDING_TOKEN
+                    token_penalty = min(1.0, token_penalty + penalty)
+
+            #Time-Based Penalty Calculation
+            if response.elapsed_time is not None:
+                time_penalty = calculate_time_penalty(
+                    elapsed_time=response.elapsed_time,
+                    limit=HARD_TIMEOUT,
```

Comment:
is wrong to define Hard_timeout, need to be used from response max_execution_time (Models)